ihmc-perception/src/main/java/us/ihmc/sensors/RealsenseImageSensor.java

```diff
@@ -105,11 +105,16 @@ protected boolean grab()
       // Update grabbed images
       synchronized (grabbedImages)
       {
+         if (grabbedImages[COLOR_IMAGE_KEY] != null)
+            grabbedImages[COLOR_IMAGE_KEY].release();
          grabbedImages[COLOR_IMAGE_KEY] = RawImage.createWithBGRImage(bgrImage,
                                                                       realsense.getColorCameraIntrinsics(),
                                                                       new FramePose3D(colorPose),
                                                                       grabTime,
                                                                       grabSequenceNumber);
+
+         if (grabbedImages[DEPTH_IMAGE_KEY] != null)
```

Comment:
Because this `grabbedImage` object is not initialized in the constructor, or at the top of the class, all these null checks are needed. This isn't something we can fix in this PR, but something to consider going forward.

Reply:
I'd prefer having a null object than a dummy image tbh

Reply:
That means when other people use the `RealsenseImageSensor` they have to put null checks everywhere. It also means you can't get the CameraIntrinsics before runtime, which in turn means you can't initialize the heightmap in the constructor. Its kind of a cascading affect the more the `RealsenseImageSensor` is used. I'm not sure we need to send a dummy image, but the null object everywhere isn't good